### PR TITLE
Tags manipulation documentation

### PR DESCRIPTION
**Pull request by ArthurHoaro**
Endpoints:

  - list all tags
  - get a tag
  - rename a tag
  - delete a tag

related to https://github.com/shaarli/Shaarli/issues/904

### CODE REVIEW

File: api-documentation.md [hunk 1]
```diff
@@ -154,6 +154,96 @@ Delete a link.
...
+
+    + Attributes (Error401)
+
++ Response 404
+
+    + Attributes (Error404)
+
+### Update a tag [PUT]
+
+Rename a tag from the given name. If the new name you provide is match an existing tag, they will be merged.
```
Comment (virtualtam) [inline api-documentation.md]: s/is match/matches/

Comment (ArthurHoaro) [reply api-documentation.md]: You is good!

File: api-documentation.md [hunk 2]
```diff
@@ -154,6 +154,96 @@ Delete a link.
...
+
+### List All Tags [GET]
+
+Retrieve a list of tags ordered by usage.
+
+An empty array will be returned if no tag is found with the filters provided. 
+
++ Parameters
+    * offset: 40 (number, optional) -  Offset from which to start listing tags (default: 0)
+    * limit: 25 (number, optional) - Number of tags to retrieve (default `all`).
```
Comment (ArthurHoaro) [inline api-documentation.md]: Missing `visibility` parameter.

File: api-documentation.md [hunk 3]
```diff
@@ -154,6 +154,96 @@ Delete a link.
...
+    + Attributes (Error400)
+
++ Response 401
+
+    + Attributes (Error401)
+
+## Tag [/tags/{tagName}]
+
++ Parameters
+    + tagName: `tutorial` (string, required) - Tag name
```
Comment (ArthurHoaro) [inline api-documentation.md:188]: Mention that it's case sensitive.

**Review by virtualtam — APPROVED** [review]

**Pull request merged**